### PR TITLE
XZ compression support

### PR DESCRIPTION
**Pull request by sesuncedu**
Adds support for XZ compression.  Slightly hacky, because the whole parser selection stuff is slightly hacky. 
Unit tests are incomplete.

### CODE REVIEW

**Comment by sesuncedu** [comment]
Also,Decompression is done every time a parser is tried.

File: api/src/main/java/org/semanticweb/owlapi/io/XZFileDocumentSource.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.semanticweb.owlapi.model.IRI;
+import org.semanticweb.owlapi.model.OWLDocumentFormat;
+import org.tukaani.xz.XZInputStream;
+
+/**
+ * An ontology document source which can read from a XZ (LZMA) compressed File.
+ * 
+ * @author ignazio
```
Comment (ignazio1977) [inline api/src/main/java/org/semanticweb/owlapi/io/XZFileDocumentSource.java]: hardly :-)

Comment (sesuncedu) [inline api/src/main/java/org/semanticweb/owlapi/io/XZFileDocumentSource.java]: By line count :)

On Thu, Mar 12, 2015 at 8:08 PM, Ignazio Palmisano <notifications@github.com

Comment (ignazio1977) [inline api/src/main/java/org/semanticweb/owlapi/io/XZFileDocumentSource.java]: Aside from the attribution that should be you, 3.4.8 is a bit misleading. Although not really a big deal.

**Comment by ignazio1977** [comment]
never heard of XZ before. Is it good?

**Comment by sesuncedu** [comment]
This is slightly unfair, but only slightly: the source files are every
revision of the Gene Ontology since 2011, converted to FSS.

snarkive:ses $ du -sh converted-ofn/
42G converted-ofn/

snarkive:ses$ ls -lh converted-ofn.tar.*
-rw-r--r-- 1 ses ses 3.9G Mar 12 10:45 converted-ofn.tar.gz
-rw-r--r-- 1 ses ses  12M Mar 12 06:22 converted-ofn.tar.xz

**Comment by ignazio1977** [comment]
I stand amazed. 400 times better than zip? Ok deal.

**Comment by sesuncedu** [comment]
It's usually not quite that good, but that happened to hit a sweet spot.
If you're using gnu tar, it's 'J' to engage :)

Compression can be slow (newer versions are multithreaded, but work on
separate blocks, so compression ratios aren't as good.

Take a look at
http://owlcs.github.io/owlapibenchmarks/horridge-iswc-2014-corpus/ for some
samples.

**Comment by sesuncedu** [comment]
Also, git repositories are much much smaller when you use sorted files
(even xml).

commit 44321553c8652028ed059ed7385657224ff8ee5f
Author: Simon Spero <sesuncedu@gmail.com>
Date:   Mon Nov 28 20:58:40 2011 -0500

    GO as it was on Mon, 28 Nov 2011 20:58:40 -0500

 go-converted.owl |    793 +-
 go.obo           |    472 +-
 go.owl           | 237232
++++++++++++++++++++++++++--------------------------
 3 files changed, 119814 insertions(+), 118683 deletions(-)

commit 4a679ebaa3a8d3c1351aff4ab9390d6e2eec7822
Author: Simon Spero <sesuncedu@gmail.com>
Date:   Tue Nov 29 11:56:26 2011 -0500

    GO as it was on Tue, 29 Nov 2011 11:56:26 -0500

 go-converted.owl |   2271 +-
 go.obo           |    673 +-
 go.owl           | 239680
++++++++++++++++++++++++++--------------------------
 3 files changed, 123888 insertions(+), 118736 deletions(-)

commit 381575066190573bb3bc3a159b5f1ca08b9c43bb
Author: Simon Spero <sesuncedu@gmail.com>
Date:   Tue Nov 29 21:16:49 2011 -0500

    GO as it was on Tue, 29 Nov 2011 21:16:49 -0500

 go-converted.owl |    662 +-
 go.obo           |    405 +-
 go.owl           | 237750
++++++++++++++++++++++++++--------------------------
 3 files changed, 119799 insertions(+), 119018 deletions(-)

commit db845227e1f880e813575c017435e342e1076381
Author: Simon Spero <sesuncedu@gmail.com>
Date:   Fri Dec 2 18:01:14 2011 -0500

    GO as it was on Fri, 02 Dec 2011 18:01:14 -0500

 go-converted.owl |    723 +-
 go.obo           |    298 +-
 go.owl           | 237986
++++++++++++++++++++++++++--------------------------
 3 files changed, 120158 insertions(+), 118849 deletions(-)

**Pull request merged**